packages/components/src/components/Modal/index.tsx

```diff
@@ -85,19 +283,36 @@ const Modal = ({
         }
     });
 
-    return (
-        <ModalContainer {...rest}>
-            <ModalWindow ref={ref} padding={padding}>
-                {cancelable && (
-                    <StyledLink onClick={onCancel}>
-                        {cancelText || ''}
-                        <StyledIcon size={16} color={colors.BLACK25} icon="CROSS" />
-                    </StyledLink>
-                )}
-                {children}
-            </ModalWindow>
-        </ModalContainer>
+    const modalWindow = (
+        <ModalWindow
+            ref={ref}
+            size={size}
+            padding={padding || getPadding(size)}
```

Comment:
just spotted that there are many `!` in ModalOverlay which is never something one would want to see :D 
isn't it possible to move || getPadding to line 256 where param destructuring is taking place, like this ...padding = getPadding(size) .. but it would also require altering Props for ModalOverlay by making the properties like padding etc required and nonNullable.. ah not sure if worth it..